openpass-sso/openpass-ios-sdk
# Initial Project Setup

## What does this PR do?

This PR establishes a file and directory structure for the project, technology requirements, and automated PR checks (via GitHub Actions) and customs.  Specifically:

* Directory Structure
  * `Development` - Development Container app
  *  `Sources` - SDK and Library modules
  *  `Tests` - Test apps and modules for testing the SDK and libraries defined in `Sources`
* README
  * Tech requirements
  * File Structure Diagram
*  GitHub
  * GitHub Action for PR tests
  * PR Rules for merging to `main`
    * Succesful PR Tests
    * Up to date branches
    * Linear History (No merge commits)
* Swift Package Manager

## Where should the reviewer start?

* `README.md`
* `Package.swift`
* `Sources/OpenPass`
* `Development/OpenPassDevelopmentApp.xcodeproj`
* `.github/workflows/test-pull-request.yml`

## Test plan

1. Open the `Development/OpenPassDevelopmentApp.xcodeproj` in Xcode
2. Run `OpenPassDevelopmentApp ` target and confirm that app launches with message: "Hello, World! This is the OpenPass SDK!"

## Screenshots / Videos

<details>
	<summary>Successful Run of OpenPassDevelopmentApp </summary>

![Simulator Screen Shot - iPhone 14 Pro Max - 2022-10-14 at 16 03 34](https://user-images.githubusercontent.com/989928/195943620-6b87a229-f764-4fe9-b64c-14f999f095e7.png)

</details>
